### PR TITLE
Update inference API explainer for model deletion

### PR DESCRIPTION
**Pull request by steveganzy**

### CODE REVIEW

File: inference-explainers/inference_api.md [hunk 1]
```diff
@@ -15,9 +15,13 @@ For Tensorflow, the SavedModel format is used while PyTorch uses the TorchScript
...
+ML models used by the inference service are fetched periodically from a linked cloud storage (e.g., [Google Cloud Storage buckets][2], [Amazon S3 buckets][3]) and made available for serving. Each model is uniquely identified by its path within the cloud storage.
+
+### Model Loading
+
+To determine the models to load, the inference service looks for a JSON model configuration file stored in the same bucket as the models. Ad techs are responsible for maintaining and updating this model configuration file. The path to the configuration file is exposed as a Terraform parameter. The B&A service periodically checks the specified configuration file (at intervals configurable by ad techs) for any changes and triggers the loading of new models into the inference service sidecar’s memory as needed.
```
Comment (akshaypundle) [inline inference-explainers/inference_api.md:24]: Can we also add that it triggers the re execution of the warm up requests as the model is loaded?

**Review by akshaypundle — APPROVED** [review]

File: inference-explainers/inference_api.md (hunk 1 above)
Comment (mihnjong-l) [inline inference-explainers/inference_api.md:24]: Can we mention "INFERENCE_MODEL_CONFIG_PATH" for the Terraform parameter we're referring to here?

File: inference-explainers/inference_api.md [hunk 2]
```diff
@@ -70,6 +74,58 @@ The optional `warm_up_batch_request_json` is a JSON batch request used to warm u
...
+    {
+      "model_path": "pcvr_v1/",
+      "checksum": "...",
+      "eviction_grace_period_in_ms": 60000,
+    },
+  ]
+}
+```
+
+#### In-Place Update
```
Comment (mihnjong-l) [inline inference-explainers/inference_api.md:125]: Can we make it "### Model In-Place Update" ?

I think CRUD (Loading, Deletion, and Update) functions should be described at the same level.

File: inference-explainers/inference_api.md [hunk 3]
```diff
@@ -70,6 +74,58 @@ The optional `warm_up_batch_request_json` is a JSON batch request used to warm u
...
+
+It is possible to update a model in place by overwriting it at the same model path in the model metadata. An in-place update is treated the same as a model deletion followed by a model addition with the previous `eviction_grace_period_in_ms` applied. This blocks the loading of new model content until the grace period expires, after which the new model content is loaded during the next polling cycle. The inference service detects this update by recognizing changes to the model checksum. When a model is updated in this way, all associated metadata, including warm_up_batch_request_json and `eviction_grace_period_in_ms`, is overwritten with the new values from the model configuration file after the previous grace period has expired and that version of the model is deleted. If the model's checksum remains unchanged between pollings, no update will be  triggered. The model along with its original metadata remains unaffected.
```
Comment (mihnjong-l) [inline inference-explainers/inference_api.md:127]: "overwriting it" -> "modifying its model entry"

File: inference-explainers/inference_api.md [hunk 4]
```diff
@@ -70,6 +74,58 @@ The optional `warm_up_batch_request_json` is a JSON batch request used to warm u
...
+
+### Model Deletion
+
+Ad techs may want to delete stale models to free up memory for the inference sidecar. This can be accomplished by removing stale model entries from the model configuration file. During the periodic fetching process, the inference service checks whether any loaded model is absent from the configuration file. If such models are found, they are deleted from the inference sidecar.
+
+For example, consider the inference service receiving its first model configuration file at time A, which instructs it to load “pcvr_v1” into the inference sidecar. At time B, if “pcvr_v1” is no longer needed and “pcvr_v2” is required instead, Ad techs can upload a new model configuration file to the cloud bucket. This operation will remove “pcvr_v1” from internal storage (since it is no longer listed in the configuration file) and attempt to load “pcvr_v2” from the cloud bucket. In other words, the inference sidecar synchronizes its internal model storage with the updated configuration file.
```
Comment (mihnjong-l) [inline inference-explainers/inference_api.md:83]: “pcvr_v1” -> `pcvr_v1` ?

Comment (mihnjong-l) [inline inference-explainers/inference_api.md:83]: nit: "instead, Ad techs can" -> "instead, ad techs can"

Comment (mihnjong-l) [inline inference-explainers/inference_api.md:83]: remove “pcvr_v1” from internal storage -> remove “pcvr_v1” from the inference sidecar ?

**Review by mihnjong-l — CHANGES_REQUESTED** [review]
Minor comments.

File: inference-explainers/inference_api.md (hunk 4 above)
Comment (mihnjong-l) [reply inference-explainers/inference_api.md]: Nit: it's not fixed. Please change

“pcvr_v1” to  \`pcvr_v1\`

“pcvr_v2” to  \`pcvr_v2\`

**Review by mihnjong-l — APPROVED** [review]
Please address my minor comment before merging.

File: inference-explainers/inference_api.md (hunk 4 above)
Comment (steveganzy) [reply inference-explainers/inference_api.md]: Done, changed here and throughout the doc

**Comment by steveganzy** [comment]
> Minor comments.

Done

**Pull request merged**